pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -373,6 +373,24 @@ static char *PyTimeToJSON(JSOBJ _obj, JSONTypeContext *tc, size_t *outLen) {
   return outValue;
 }
 
+static char *PyDecimalToUTF8Callback(JSOBJ _obj, JSONTypeContext *tc,
+                                     size_t *len) {
+  PyObject *obj = (PyObject *)_obj;
+  PyObject *format_spec = PyUnicode_FromStringAndSize("f", 1);
+  PyObject *str = PyObject_Format(obj, format_spec);
+  Py_DECREF(format_spec);
+
+  if (str == NULL) {
+    *len = 0;
+    ((JSONObjectEncoder *)tc->encoder)->errorMsg = "";
+    return NULL;
+  }
+
+  GET_TC(tc)->newObj = str;
+  char *outValue = (char *)PyUnicode_AsUTF8AndSize(str, (Py_ssize_t *)len);
```

Comment:
```suggestion
  Py_ssize_t s_len = len;
  char *outValue = (char *)PyUnicode_AsUTF8AndSize(str, &s_len);
```

At the risk of being pedantic, it might be worth following rules for strict aliasing here.

Of course there's still also the issue of casting between `char *` and `const char*` although that's out of scope for this particular issue...